packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreen.kt

```diff
@@ -0,0 +1,80 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.util.Log
+import org.unimodules.core.interfaces.SingletonModule
+import java.util.*
+
+object SplashScreen: SingletonModule {
+  private const val TAG = "SplashScreen"
+
+  override fun getName(): String {
+    return "SplashScreen"
+  }
+
+  private val controllers = WeakHashMap<Activity, SplashScreenController>()
+
+  /**
+   * Show SplashScreen by mounting it in ContentView.
+   * @param activity                 Target Activity for SplashScreen to be mounted in.
+   * @param resizeMode               SplashScreen imageView resizeMode.
+   * @param rootViewClass            Class of View that would be monitored for children occurence (autohiding feature).
+   * @param splashScreenResourcesProvider
+   * @param successCallback          Callback to be called once SplashScreen is mounted in view hierarchy.
+   * @param failureCallback          Callback to be called once SplashScreen cannot be mounted.
+   * @throws [expo.modules.splashscreen.exceptions.NoContentViewException] when [SplashScreen.show] is called before [Activity.setContentView] (when no ContentView is present for given activity).
+   */
+  @JvmStatic
+  @JvmOverloads
+  fun show(
+      activity: Activity,
+      resizeMode: SplashScreenImageResizeMode,
+      rootViewClass: Class<*>,
+      splashScreenResourcesProvider: SplashScreenResourcesProvider = NativeResourcesBasedSplashScreenResourcesProvider(),
+      successCallback: () -> Unit = {},
+      failureCallback: (reason: String) -> Unit = { Log.w(TAG, it) }
+  ) {
+    // SplashScreen.show can only be called once per activity
+    if (controllers.containsKey(activity)) {
+      return failureCallback("'SplashScreen.show' has already been called for this activity.")
```

Comment:
As for me it writing
```kotlin
failureCallback(...)
return
```
is as readable as
```kotlin
return failureCallback(...)
```

I just prefer not using this dangling `return` statements, because it my eyes it break some nicely written (functional-inspired) flow 😉